content/en/blog/2019/knative-activator-adapter/index.md

```diff
@@ -0,0 +1,94 @@
+
+The following diagram represents the current Knative architecture for scale-from-zero.
+
+{{< image width="60%" link="knative-activator.png" caption="Knative scale-from-zero" >}}
+
+The traffic for an idle application is redirected to **Activator** component by programming Istio with `VirtualServices`
+and `DestinationRules`. When **Activator** receives a new request, it:
+
+1. buffers incoming requests
+1. triggers the **Autoscaler**
+1. redirects requests to the application after it has been scaled up, including retries and load-balancing (if needed)
+
+Once the application is up and running again, Knative restores the routing from **Activator** to the running application.
+
+## Mixer adapter
+
+[Mixer](https://istio.io/faq/mixer/) provides a rich intermediation layer between the Istio components and infrastructure backends.
+It is designed as a stand-alone component, separate from [Envoy](https://www.envoyproxy.io/), and has a simple extensibility model
+to enable Istio to interoperate with a wide breadth of backends. Mixer is inherently easier to extend
+than Envoy is.
+
+Mixer is an attribute processing engine that uses operator-supplied configuration to map request attributes from the Istio proxy into calls
+to the infrastructure backends systems via a pluggable set of adapters. Adapters enable **Mixer** to expose a single consistent API, independent of the
+infrastructure backends in use. The exact set of adapters used at runtime is determined through operator configuration and can easily
+be extended to target new or custom infrastructure backends.
+
+Here you utilized these benefits to implement the Knative scale-from-zero logic in a special Mixer
+[out-of-process adapter](https://github.com/istio/istio/wiki/Mixer-Out-Of-Process-Adapter-Dev-Guide).
+Prior Istio releases required building the custom adapter along with the istio proxy. The out-of-process adapter 
+allows you to use any programming language and to build and maintain your extension as a stand-alone
+program without the need to build the Istio proxy.
+
+The following diagram represents the Knative design using the **Mixer** adapter.
+
+{{< image width="60%" link="knative-mixer-adapter.png" caption="Knative scale-from-zero" >}}
+
+In this design, there is no need to change the routing from/to **Activator** for an idle application as in the original Knative setup.
+When the Istio proxy represented by the ingress gateway component receives a new request for an idle application, it informs **Mixer**, including all the
+relevant metadata information.
+**Mixer** then calls your adapter which triggers the Knative **Autoscaler** using the original Knative protocol.
+{{< idea >}}
+By using this desgin you do not need to deal with buffering, retries and load-balancing because it is already handled by the Istio proxy.
+{{< /idea >}}
+
+Istio's capabilities to extend Mixer makes it very simple to convert a complex networking-based application logic into a few lines
```

Comment:
Can we make this customer-focused, rather than developer-focused? Specifically, I suggest getting rid of the Go code and replacing the text with:

Istio's use of Mixer adapters makes it possible to replace otherwise complex networking-based application logic with a more straightforward implementation, as demonstrated in [the `kactivator` adapter](https://github.com/zachidan/istio-kactivator).